src/common/utils.cc

```diff
@@ -0,0 +1,61 @@
+// Copyright (C) 2019-2023 Zilliz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under the License.
+
+#include "knowhere/utils.h"
+
+#include <cmath>
+#include <cstdint>
+
+#include "knowhere/comp/time_recorder.h"
+#include "knowhere/log.h"
+
+namespace knowhere {
+
+const float floatDiff = 0.00001;
+
+void
+normalize(float* data, int32_t dim) {
```

Comment:
Pretty confusing with this signature. Since it claims to normalize vectors (because we have dim) without rows.
Suggestions:
1. Change the dim to num.
2. Change the normalize to NormalizeVec (or some other names), and data to vector.
3. Add a new parameter called row and default it by 1. For loop normalize `data` as a list of vectors.
Let's pick one of the ways to fix it.

Reply:
updated